File: backend/Domain/Models/Produto.cs
using MarketPlace.Domain.Models.DTOs;
using System.ComponentModel.DataAnnotations.Schema;
using MarketPlace.Interfaces.IGeneric;

namespace MarketPlace.Domain.Models;

public class Produto : IResultSearch
{
    public Guid Id { get; set; }
    public string Nome { get; set; }
    public string Descricao { get; private set; }
    public decimal Preco { get; private set; }
    public int Estoque { get; private set; }

    public Guid LojaId { get; private set; }
    [ForeignKey("LojaId")] public Loja? Loja => null;

    public Produto(string nome, string descricao, decimal preco, int estoque, Guid lojaId)
    {
        Nome = nome ?? throw new ArgumentNullException(nameof(nome));
        Descricao = descricao ?? throw new ArgumentNullException(nameof(descricao));
        Preco = preco;
        Estoque = estoque;
        LojaId = lojaId;
    }

    public Produto(ProdutoDto produtoDto)
    {
        Nome = produtoDto.Nome;
        Descricao = produtoDto.Descricao;
        Preco = produtoDto.Preco;
        Estoque = produtoDto.Estoque;
        LojaId = produtoDto.LojaId;
    }
}

File: frontend/src/app/domain/models/produto.ts
import { Loja } from "./loja";

export class Produto {
  id: string;
  nome: string;
  descricao: string;
  estoque: number;
  preco: number;
  imagemUrl: string;
  lojaId: string;
  loja?: Loja;

  constructor(
    id: string,
    nome: string,
    descricao: string,
    estoque: number,
    preco: number,
    imagemUrl: string,
    lojaId: string,
  ) {
    this.id = id;
    this.nome = nome;
    this.descricao = descricao;
    this.preco = preco;
    this.estoque = estoque;
    this.imagemUrl = imagemUrl;
    this.lojaId = lojaId;
  }
}
